### ISSUE TITLE
Feature request: Inverted conveyor filters

### ISSUE THREAD

**Issue by 7eggert** [cyanobacterium/PowerAdvantageAPI#17]
I suggest inverted conveyors: The normal conveyors with their logic reversed.

Either this could be done by having an inventory slot accepting a redstone torch or (maybe) a switch (if filled, then invert logic) or by crafting a special variant: Conveyor + RS torch -> inverted conveyor.

**Comment by DrPlantabyte** [comment]
What about using a redstone signal to invert the logic?

**Comment by 7eggert** [comment]
I don't like that. It would require a RS signal within a constrained area - at least 3 of the directions are used for i/o, rs would be the fourth. Also there may be stray signals from turning off machines.

OTOH, it might be useful ...

Maybe with a slot: Empty = normal; rs torch = inverted; redstone dust = by signal.

**Comment by DrPlantabyte** [comment]
I don't think that is intuitive enough for the average user, so we'll need another solution.
How about a sorter block with 6 inventory slots that can only hold conveyor blocks (one for each face). When an item is put into the sorter, it will push the item out of the face whose filter it matches (a normal conveyor means "match any" in this case).

**Comment by 7eggert** [comment]
I'd stick with just inverting the logic then. that's enough for most things.

If you want to have an external signal, maybe it can be changed to be obvious: Add a rs dust / rs torch symbol, print "not inverted" / "inverted on redstone" / "always inverted" next to the slot.

If you'd like a more capable sorter (I do): The (IIRC) RFTools mod has a block that will only accept certain items from certain directions, with 9 internal slots - but it's a passive block and it can't blacklist, which limits it's uses. That combined with a conveyor and blacklists would be a dream and the swiss army knife of item sorting and item tranport.

**Comment by DrPlantabyte** [comment]
Adding another 8 filtered conveyor blocks is pretty drastic just to swap the down and sideways directions.

How about you describe a use-case where this is important so I can understand the context for this request (I truly don't get it) and we'll come up with a solution that minimizes the development time and chances for bugs.

**Comment by 7eggert** [comment]
I set up a lot of automated filters behind a front, and this front is already too big. Therefore it's a little bit too crowded behind that front. Being able to reverse the filter would safe some space.

I'd want to deliver swords and hoes upwards to my farms, wood and burnables down excepte.g. chests, rest return to a different chest. right next to that, I'd sort ores, plants, poisonous potatoes and steaks. The if-then-else is a little bit complicated when you cant't easily reverse the logic.

**Comment by DrPlantabyte** [comment]
In that case, I think that a 6-sided sorter block will be more useful to you. Then you could use it as a simple inverter or to essentially cram several filters into the space of a single block.

**Comment by jriwanek** [comment]
I like the sorter idea personally.

**Comment by 7eggert** [comment]
I assumed a sorter would be overpowered and NACKed, but I like it. Especially if it can sort by is_not_ore() etc.
